Source/RunActivity/Viewer3D/MSTSSky.cs

```diff
@@ -209,58 +210,59 @@ public void PrepareFrame(RenderFrame frame, ElapsedTime elapsedTime)
                 }
             }
 
-            ////////////////////////////////////////////////////////////////////
-
-            // Current solar and lunar position are calculated by interpolation in the lookup arrays.
-            // Using the Lerp() function, so need to calculate the in-between differential
-            float diff = (float)(MSTSSkyViewer.Simulator.ClockTime - mstsskyoldClockTime) / 1200;
-            // The rest of this increments/decrements the array indices and checks for overshoot/undershoot.
-            if (MSTSSkyViewer.Simulator.ClockTime >= (mstsskyoldClockTime + 1200)) // Plus key, or normal forward in time
-            {
-                step1++;
-                step2++;
-                mstsskyoldClockTime = MSTSSkyViewer.Simulator.ClockTime;
-                diff = 0;
-                if (step2 >= maxSteps) // Midnight.
-                {
-                    step2 = 0;
-                }
-                if (step1 >= maxSteps) // Midnight.
-                {
-                    step1 = 0;
-                }
-            }
-            if (MSTSSkyViewer.Simulator.ClockTime <= (mstsskyoldClockTime - 1200)) // Minus key
-            {
-                step1--;
-                step2--;
-                mstsskyoldClockTime = MSTSSkyViewer.Simulator.ClockTime;
-                diff = 0;
-                if (step1 < 0) // Midnight.
-                {
-                    step1 = maxSteps - 1;
-                }
-                if (step2 < 0) // Midnight.
-                {
-                    step2 = maxSteps - 1;
-                }
-            }
-
-
-            mstsskysolarDirection.X = MathHelper.Lerp(mstsskysolarPosArray[step1].X, mstsskysolarPosArray[step2].X, diff);
-            mstsskysolarDirection.Y = MathHelper.Lerp(mstsskysolarPosArray[step1].Y, mstsskysolarPosArray[step2].Y, diff);
-            mstsskysolarDirection.Z = MathHelper.Lerp(mstsskysolarPosArray[step1].Z, mstsskysolarPosArray[step2].Z, diff);
-            mstsskylunarDirection.X = MathHelper.Lerp(mstsskylunarPosArray[step1].X, mstsskylunarPosArray[step2].X, diff);
-            mstsskylunarDirection.Y = MathHelper.Lerp(mstsskylunarPosArray[step1].Y, mstsskylunarPosArray[step2].Y, diff);
-            mstsskylunarDirection.Z = MathHelper.Lerp(mstsskylunarPosArray[step1].Z, mstsskylunarPosArray[step2].Z, diff);
+            //////////////////////////////////////////////////////////////////////
+
+            //// Current solar and lunar position are calculated by interpolation in the lookup arrays.
+            //// Using the Lerp() function, so need to calculate the in-between differential
+            //float diff = (float)(MSTSSkyViewer.Simulator.ClockTime - mstsskyoldClockTime) / 1200;
+            //// The rest of this increments/decrements the array indices and checks for overshoot/undershoot.
+            //if (MSTSSkyViewer.Simulator.ClockTime >= (mstsskyoldClockTime + 1200)) // Plus key, or normal forward in time
+            //{
```

Comment:
Thank you Chris for your efforts, I think this is absolutely fine. For now it is just important that the sky and mstssky don't diverge more, because later noone would know which is the corrected and which is the old part. However I think instead of commenting out code sections it is better just to delete these lines. The old will always remain available in git anyway, so there is no need to keep theese lines in the live code.